plugin/core/registry.py

```diff
@@ -104,31 +104,24 @@ def sessions(self, capability: Optional[str] = None) -> Generator[Session, None,
         yield from sessions_for_view(self.view, capability)
 
 
-class LspRestartAllServersCommand(sublime_plugin.TextCommand):
-    def is_enabled(self) -> bool:
-        return is_supported_view(self.view)
-
-    def run(self, edit: Any) -> None:
-        window = self.view.window()
-        if window:
-            windows.lookup(window).restart_sessions()
-
-
 class LspRestartServerCommand(sublime_plugin.TextCommand):
     def is_enabled(self) -> bool:
-        return is_supported_view(self.view) and windows.lookup(self.view.window()).has_active_sessions()
+        return any(self.sessions())
 
     def run(self, edit: Any) -> None:
         window = self.view.window()
         if not window:
             return
 
-        sessions = windows.lookup(window).get_sessions()
+        sessions = [session.config.name for session in self.sessions()]
         if sessions:
             window.show_quick_panel(sessions, lambda index: self.restart_server(index, sessions))
 
     def restart_server(self, index, sessions) -> None:
         if index == -1:
             return
-        windows.lookup(self.view.window()).restart_session(config_name=sessions[index])
+        windows.lookup(self.view.window()).restart_session_async(config_name=sessions[index])
         return
+
+    def sessions(self, capability: Optional[str] = None) -> Generator[Session, None, None]:
```

Comment:
You can inherit from LspTextCommand which will give you `is_enabled` and `sessions` implementations already.